pallets/valve/src/lib.rs

```diff
@@ -86,26 +104,68 @@ pub mod pallet {
 
 			ValveClosed::<T>::put(true);
 			<Pallet<T>>::deposit_event(Event::ValveClosed);
-			Ok(().into())
+			Ok(())
+		}
+
+		/// Tap the valve.
+		///
+		/// If valve is closed you cannot tap a pallet.
+		/// You cannot tap this paller, as then you could never untap it.
+		#[pallet::weight(T::DbWeight::get().read + 2 * T::DbWeight::get().write)]
+		pub fn tighten_valve(origin: OriginFor<T>, pallet_name: Vec<u8>) -> DispatchResult {
+			ensure_root(origin)?;
+
+			// Ensure the valve isn't already off.
+			// If the valve is already off there is no need to tap individual pallets.
+			ensure!(!ValveClosed::<T>::get(), Error::<T>::ValveAlreadyClosed);
+
+			let pallet_name_string =
+				sp_std::str::from_utf8(&pallet_name).map_err(|_| Error::<T>::InvalidCharacter)?;
+
+			// Not allowed to pause calls of this pallet as then you could never unpause them.
+			ensure!(
+				pallet_name_string != <Self as PalletInfoAccess>::name(),
+				Error::<T>::CannotPause
+			);
+
+			TappedPallets::<T>::mutate_exists(pallet_name.clone(), |maybe_tapped| {
+				if maybe_tapped.is_none() {
+					*maybe_tapped = Some(());
+					Self::deposit_event(Event::PalletTapped { pallet_name_bytes: pallet_name });
+				}
+			});
+
+			Ok(())
 		}
 
 		/// Return the chain to normal operating mode.
 		///
 		/// Weight cost is:
 		/// * One DB read to ensure the valve is closed.
-		/// * Two DB writes - 1 for the mode and 1 for the event.
-		#[pallet::weight(T::DbWeight::get().read + 2 * T::DbWeight::get().write)]
+		/// * Three DB writes - 1 for the mode and 1 for the event.
+		#[pallet::weight(T::DbWeight::get().read + 3 * T::DbWeight::get().write)]
 		pub fn open_valve(origin: OriginFor<T>) -> DispatchResult {
 			ensure_root(origin)?;
 
-			// Ensure the valve is off.
-			// This test is not strictly necessary, but seeing the error may help a confused chain
-			// operator during an emergency
-			ensure!(ValveClosed::<T>::get(), Error::<T>::ValveAlreadyOpen);
-
 			ValveClosed::<T>::put(false);
+			TappedPallets::<T>::remove_all(None);
+
 			<Pallet<T>>::deposit_event(Event::ValveOpen);
-			Ok(().into())
+			Ok(())
+		}
+
+		/// Untap the pallet.
+		#[pallet::weight(T::DbWeight::get().read + 2 * T::DbWeight::get().write)]
+		pub fn loosen_valve(origin: OriginFor<T>, pallet_name: Vec<u8>) -> DispatchResult {
+			ensure_root(origin)?;
+
+			// If the valve is off then you cannot lossen it.
+			ensure!(!ValveClosed::<T>::get(), Error::<T>::ValveAlreadyClosed);
```

Comment:
nit: i thought no camel case?

Reply:
also, do you mean if valve is open you can't loosen it? cause if a valve (in life) is closed, you can def loosen it haha

Reply:
camel case for types not for var and fn names

Reply:
This is where the names are weird. If you close the valve it means you locked down the chain essentially.

If you tap a pallet it means you shut off that pallet. 

It doesn't matter if you try to untap a pallet if the chain is closed. It won't do anything